build.gradle

```diff
@@ -225,130 +172,204 @@ configure(subprojects - [project(":spring-cloud-starter-app-broker"),
 	javadoc {
 		description = "Generates project-level javadoc for use in -javadoc jar"
 
-		options.memberLevel = org.gradle.external.javadoc.JavadocMemberLevel.PROTECTED
+		options.memberLevel = JavadocMemberLevel.PROTECTED
 		options.author = true
 		options.header = project.name
 		options.links(javadocLinks)
 		options.addStringOption('Xdoclint:none', '-quiet')
 	}
 
-	task sourcesJar(type: Jar, dependsOn:classes) {
+	sourcesJar {
 		duplicatesStrategy = DuplicatesStrategy.EXCLUDE
-		archiveClassifier = "sources"
+		archiveClassifier.set "sources"
 		from sourceSets.main.allJava
 	}
 
-	task javadocJar(type: Jar) {
-		archiveClassifier = "javadoc"
+	javadocJar {
+		duplicatesStrategy = DuplicatesStrategy.EXCLUDE
+		archiveClassifier.set "javadoc"
 		from javadoc
 	}
+}
+
+configure(staticAnalysisProjects) {
+	apply plugin: "checkstyle"
+	apply plugin: "pmd"
+	apply plugin: "jacoco"
 
-	artifacts {
-		archives sourcesJar
-		archives javadocJar
+	checkstyle {
+		configFile = file("${project.rootDir}/src/checkstyle/checkstyle.xml")
+		toolVersion = "${checkstyleVersion}"
+		showViolations = true
+	}
+	checkstyleMain {
+		source = "src/main/java"
+	}
+	checkstyleTest {
+		source = "src/test/java"
 	}
 
-	configurations {
-		// exclude JUnit 4 globally, in favor of JUnit 5
-		testImplementation.exclude group: "junit", module: "junit"
+	pmd {
+		toolVersion = "${pmdVersion}"
+		consoleOutput = true
+	}
+	pmdMain {
+		ruleSets = []
+		ruleSetFiles = files("${project.rootDir}/src/pmd/pmdRuleSet.xml")
+		source = "src/main/java"
+	}
+	pmdTest {
+		ruleSets = []
+		ruleSetFiles = files("${project.rootDir}/src/pmd/pmdTestRuleSet.xml")
+		source = "src/test/java"
 	}
 }
 
-configure(rootProject) {
-	description = "Spring Cloud App Broker"
+configure(publishedProjects) {
+	apply plugin: 'maven-publish'
 
-	// don't publish the default jar for the root project
-	configurations.archives.artifacts.clear()
+	publishing {
+		repositories {
+			maven {
+				url = project.properties.getOrDefault('publicationRepository', "${System.getenv('HOME')}/.m2/repository")
+			}
+		}
+		publications {
+			mavenJava(MavenPublication) {
+				suppressAllPomMetadataWarnings()
+				from components.java
+				pom {
+					afterEvaluate {
+						name = project.description
+						description = project.description
+					}
+					packaging = starterProjects.contains(project) ? "pom" : null
```

Comment:
oh I see. you can ignore my other comments about pom packaging in the starters if this works as I'm reading it.